### PR TITLE
Cleanup of layout to make <landing-page> less contextually specific

### PR DESCRIPTION
**Pull request by rhamilto**
changes include:

* switches layout to use flex (eliminates fixed positioning since that
could be problematic when <landing-page> is put in other contexts (like
origin-web-console))
* moves app-specific styling to app.less since landing-page.less should
be contextually agnostic (the app will need to control “advanced”
layout features like inner scrolling)
* the panel now slides down in front of the overlay

known issues:

* the layout of the content on the landing page is a bit janky, but
@cshinn is working on that in parallel to this effort
* the snap-up icon isn’t fully visible on clicking snap-down, but I
think the entire mechanism needs rethinking now that @spadgett changed
the layout the cards/tiles

### CODE REVIEW

File: src/styles/animations.less [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    margin-top: -50%;
+  }
+  100% {
+    margin-top: 0%;
+    opacity: 1;
+  }
+}
+
+@keyframes modalSlideDown {
+  0% {
```
Comment (dtaylor113) [inline src/styles/animations.less]: Do we need all four of these annimations?  I'm not sure the 'simple' animations are being used.
I see in overlay-panel.less:
```
.catalogs-overlay-window {
  &.ng-enter {
    animation: simpleSlideDown 1s ease forwards;
  }
  &.ng-leave {
    animation: simpleSlideUp 1s ease forwards;
  }
}
```
However, I don't see the class 'catalogs-overlay-window' used anywhere, so we maybe able to get rid of the 'simple' animations.

Comment (rhamilto) [inline src/styles/animations.less]: Good catch, @dtaylor113!  I've removed the animations and the rules as they weren't in use.

**Comment by jeff-phillips-18** [comment]
The design of the landing page is for the search component and the side bar to be fixed. @serenamarie125 @beanh66

**Comment by jeff-phillips-18** [comment]
Shouldn't there be a left margin on the page? The text runs right up against the window edge.

**Comment by jeff-phillips-18** [comment]
The edit and create project panels no longer show the panel titles or field labels.

**Comment by jeff-phillips-18** [comment]
The edit and create project panels are fixed but the sidebar isn't. If the user scrolls, the panel does not move with the target it points to.

**Comment by rhamilto** [comment]
> The design of the landing page is for the search component and the side bar to be fixed. @serenamarie125 @beanh66

@jeff-phillips-18, that position fixing needs to be done at the app level, which I can add back to this demo app, but it will need to be addressed in each context (app) that utilizes <landing-page> since position: fixed is relative to the viewport, and we won't know what's at the top, right, left, and bottom of the viewport.  For example, when <landing-page> gets put in to origin-web-console, the offset properties are going to vary from what's in this demo app since the navbar height is going to differ (and is even customized by customers in some instances).

**Comment by rhamilto** [comment]
> Shouldn't there be a left margin on the page? The text runs right up against the window edge.

We'll fix that when we style the content.  @cshinn is working on the apps section in parallel to this effort.

**Comment by rhamilto** [comment]
> The edit and create project panels no longer show the panel titles or field labels.

Sorry.  I meant to document that as a known issue.  That'll get fixed when I style the sidebar stuff (which is blocked by these layout change needs).

**Comment by spadgett** [comment]
> The design of the landing page is for the search component and the side bar to be fixed. 

What happens if the content of the sidebar is too long to fit on the screen?

**Comment by spadgett** [comment]
> What happens if the content of the sidebar is too long to fit on the screen?

To be clear, I'm not saying it shouldn't be fixed, but there are some problems we'll have to solve like different banner heights in different contexts (as @rhamilto points out) and content that overflows.

**Comment by rhamilto** [comment]
> The edit and create project panels are fixed but the sidebar isn't. If the user scrolls, the panel does not move with the target it points to.

Missed that one since I wasn't focused on the contents.  Lemme fix.

**Comment by rhamilto** [comment]
>> The edit and create project panels are fixed but the sidebar isn't. If the user scrolls, the panel does not move with the target it points to.

> Missed that one since I wasn't focused on the contents. Lemme fix.

Done.

**Comment by rhamilto** [comment]
Ok.  I believe I've fixed all the noted issues.  PTAnotherL.

File: src/styles/landing-page.less [hunk 2]
```diff
@@ -1,88 +1,34 @@
...
-
 .landing-main-area {
-  margin: 15px 20px;
-  padding-right: 10px;
-
-  // TODO: Remove corresponding style in services-view.less (card-expansion)
-  // when sidebar is no longer fixed position.
-  @media (min-width: @screen-sm-min) {
-    margin-right: @sidebar-width;
+  padding: (@grid-gutter-width / 2); // this is temp
```
Comment (rhamilto) [inline src/styles/landing-page.less]: @cshinn, you're gonna want to remove this to get the styling of the search and apps area right, but I ended up putting it in to make the page look ok in the interim.

Comment (cshinn) [inline src/styles/landing-page.less]: Thanks for the heads up!

**Pull request merged**